### ISSUE TITLE
Improved notification control on Desktop

### ISSUE THREAD

**Issue by bjtrbjtr** [klinker-apps/messenger-issues#339]
**Device with issue:** Dell XPS 15 9560 running Windows 10
**Primary device:** Samsung Galaxy S5 (SM-G900V) running Android 6.0.1
**App version:** 2.3.2.1207
**Desktop modality:** Using Chrome App

**Request:**

It would be great to have improved control over desktop notifications when SMS messages arrive, distinct from the conversation-level control settings available on the Android phone app. Right now, a notification always appears, and the notification content includes a preview of the SMS content. The requested features are:

1. It would be great to have a "snooze desktop notifications" global option (e.g., snooze for NN minutes, or snooze until HH:MM time). I frequently teach or present from my computer, and want to avoid notification pop-ups then.
2. It would also be great to have a global no-content-preview option for the desktop, so you can get a notification that you have a message, but perhaps not the content, regardless of whether the conversation is marked as private or not. Same use scenario as above.

Thanks for this great product!

**Comment by klinker24** [comment]
Thanks for the requests! 

The first and second are two that I would definitely like to do. I will probably add a separate settings screen for this, that isn't synced with the phone, so I guess that is kind of what you are suggesting for number 3 as well.

**Comment by bjtrbjtr** [comment]
Awesome--thank you!

**Comment by klinker24** [comment]
I added a toggle in the chrome extension's settings to disable content in notifications: https://github.com/klinker41/messenger-chrome/commit/d27acb9d2b5db7ad21eb0877ca2494639a982d4c

It should be available now, in `v1.9.0` of the extension

**Comment by bjtrbjtr** [comment]
Thank you! That sounds great!
Ben

**Comment by klinker24** [comment]
I added this to the `1.1.0` milestone for the desktop apps. I don't think that the new functionality will come to the Chrome app/extension though. These will be improvements for the desktop applications only. Notifications in general just aren't consistent enough on the Chrome platform to continue suggesting them over the desktop applications.

To finish this, I would like to:
1. Add snooze option within the desktop app
2. Add option to not display content previews within the desktop app

Along with this, I want to redo the `Preference` menu option on the desktop apps.

```
Show in menu bar
Notification preferences >
  Show notifications
  Show unread count on icon
  Show message preview in notifications
  ---
  Snooze desktop notifications >
    30 mins
    1 hour
    3 hours
    12 hours
```

**Comment by klinker24** [comment]
This is now finished for the native desktop apps. It will be released with v1.2.0 of the desktop apps.
